cloudevents/conversion.py

```diff
@@ -222,3 +240,62 @@ def to_binary(
         format=converters.TypeBinary,
         data_marshaller=data_marshaller,
     )
+
+
+def encode_attribute_value(value: typing.Any) -> typing.Any:
+    if isinstance(value, enum.Enum):
+        return value.value
+    if isinstance(value, datetime.datetime):
+        return value.isoformat()
+
+    return value
+
+
+def from_dict(
+    event_type: typing.Type[AnyCloudEvent],
+    event: typing.Dict[str, typing.Any],
+) -> AnyCloudEvent:
+    """
+    Construct an Event object from a dict event representation
+    :param event: Event represented via dict
+    :param event_type: Type of the event to construct from the dict
+    """
+    return event_type.create(
+        {
+            attr_name: encode_attribute_value(attr_value)
+            for attr_name, attr_value in event.items()
+            if attr_name != "data"
+        },
+        data=event.get("data"),
+    )
+
+
+def to_dict(event: AnyCloudEvent) -> typing.Dict[str, typing.Any]:
+    """
+    Convert an event to a simple dict representation
+    :param event: Event you wish to convert into a dict.
+    """
+    result = {attribute_name: event.get(attribute_name) for attribute_name in event}
+    result["data"] = event.data
+    return result
+
+
+def _json_or_string(
+    content: typing.Optional[typing.AnyStr],
+) -> typing.Optional[
+    typing.Union[
+        typing.Dict[typing.Any, typing.Any],
+        typing.List[typing.Any],
+        typing.AnyStr,
+    ]
+]:
+    """
+    Given an encoded JSON string MUST return decoded JSON object.
+    Otherwise, MUST return the given string as-is.
+    """
```

Comment:
```suggestion
    """
    Returns a JSON-decoded dictionary or a list of dictionaries if a valid JSON string is provided.
    
    Returns the same `content` in case of an error or `None` when no content provided.
    """
```